tracing-subscriber/src/filter/env/directive.rs

```diff
@@ -692,6 +710,13 @@ impl SpanMatcher {
             .unwrap_or_else(|| self.base_level.clone())
     }
 
+    pub(crate) fn target(&self) -> Option<Arc<str>> {
+        self.field_matches
```

Comment:
Why not just put the `target` field on `SpanMatcher`?

Reply:
So not sure what is right here. I went with the first target, which for me will work.

Reply:
Sorry if that was unclear — what I was saying is, it doesn't really seem necessary for every `SpanMatch` to have a target field, when we can just clone the `CallsiteMatch`'s target when constructing a new `SpanMatcher`, here:
https://github.com/tokio-rs/tracing/blob/019d61e897088010ba8a6f64d37d9cc97e4dca49/tracing-subscriber/src/filter/env/directive.rs#L673-L687

We could change that function to:
```rust
 pub(crate) fn to_span_match(&self, attrs: &span::Attributes<'_>) -> SpanMatcher { 
     let field_matches = ...;
     SpanMatcher { 
         field_matches, 
         base_level: self.base_level.clone(), 
         target: self.target.clone();
     } 
 } 
```
and remove the `target` field from `SpanMatch`.